From d716e636231c8d636bf2139896d817b66fe6d510 Mon Sep 17 00:00:00 2001
From: Marek Kasik <mkasik@redhat.com>
Date: Thu, 21 Mar 2019 13:15:37 +0100
Subject: [PATCH 1/2] cairo: Compute correct coverage values for box filter

Use double precision for computation of coverage
of the left most pixel in the box filter.

Issue #736
---
 poppler/CairoRescaleBox.cc | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/poppler/CairoRescaleBox.cc b/poppler/CairoRescaleBox.cc
index b8371a5b..d7615010 100644
--- a/poppler/CairoRescaleBox.cc
+++ b/poppler/CairoRescaleBox.cc
@@ -226,10 +227,10 @@ static int compute_coverage (int coverage[], int src_length, int dest_length)
     /* I have a proof of this, which this margin is too narrow to contain */
     for (i=0; i<dest_length; i++)
     {
-        float left_side = i*scale;
-        float right_side = (i+1)*scale;
-        float right_fract = right_side - floor (right_side);
-        float left_fract = ceil (left_side) - left_side;
+        double left_side = i*scale;
+        double right_side = (i+1)*scale;
+        double right_fract = right_side - floor (right_side);
+        double left_fract = ceil (left_side) - left_side;
         int overage;
         /* find out how many source pixels will be used to fill the box */
         int count = floor (right_side) - ceil (left_side);
-- 
2.20.1

